src/main/java/de/topobyte/jts/utils/PolygonHelper.java

```diff
@@ -138,12 +139,9 @@ public static MultiPolygon multipolygonFromRings(Set<LinearRing> rings,
 			candidates.addAll(rings);
 			candidates.remove(r);
 			Polygon p1 = ringToPolygon.get(r);
-			for (LinearRing c : candidates) {
-				Polygon p2 = ringToPolygon.get(c);
-				if (p1.contains(p2)) {
-					graph.addEdge(r, c);
-				}
-			}
+			List<LinearRing> containedRings = candidates.parallelStream().
+					filter(c -> p1.contains(ringToPolygon.get(c))).collect(Collectors.toList());
```

Comment:
I guess now that we no longer have a variable `p2`, we could as well rename `p1` to just `p`. Would you mind doing that as part of this pull request?